### PR TITLE
div

### PR DESCRIPTION
**Pull request by hellozmz**
## Description

- diopiDivInpScalar
- diopiDivScalar
- diopiDiv
- diopiDivInp use op_api

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/ascend_npu/diopi_impl/binary.cpp [hunk 1]
```diff
@@ -6,9 +6,67 @@
...
+    std::string mode = roundModeStr(roundingMode);
+    if (mode.empty()) {
+        op_api::div_(inputAt, otherAt);
+    } else {
+        op_api::div_(inputAt, otherAt, mode);
+    }
+    END_CALL_ACL_OP();
+}
+
+diopiError_t diopiDivScalar(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input, const diopiScalar_t* other,
```
Comment (yangbofun) [inline impl/ascend_npu/diopi_impl/binary.cpp:47]: 1. 这里实现了diopiDivScalar， 是不是上面的impl/ascend_npu/ascend_config.yaml里，把diopiDivScalar 改为ascend_npu.
2. diopiDivInpScalar同理。

Comment (hellozmz) [reply impl/ascend_npu/diopi_impl/binary.cpp:47]: diopiDivScalar / diopiDivInpScalar 还没有调试通过，暂时没有放到ascend_npu中。

File: impl/ascend_npu/diopi_impl/binary.cpp [hunk 2]
```diff
@@ -6,9 +6,69 @@
...
+                            diopiRoundMode_t roundingMode) {
+    BEGIN_CALL_ACL_OP(input, other, out);
+    std::string mode = roundModeStr(roundingMode);
+    if (mode.empty()) {
+        at::Tensor result = op_api::div(inputAt, otherAt);
+        outAt.copy_(result);
+    } else {
+        at::Tensor result = op_api::div(inputAt, otherAt, mode);
+        outAt.copy_(result);
```
Comment (zhaoguochun1995) [inline impl/ascend_npu/diopi_impl/binary.cpp:56]: at::Tensor & div_out(const at::Tensor & self, const at::Tensor & other, at::Tensor & out);
at::Tensor & div_out(const at::Tensor & self, const at::Tensor & other, c10::optional<c10::string_view> rounding_mode, at::Tensor & out);
用out版本可以减少一次copy

Comment (zhaoguochun1995) [reply impl/ascend_npu/diopi_impl/binary.cpp:56]: at::Tensor & div_(at::Tensor & self, const at::Scalar & other);
at::Tensor & div_(at::Tensor & self, const at::Scalar & other, c10::optional<c10::string_view> rounding_mode);

Comment (zhaoguochun1995) [reply impl/ascend_npu/diopi_impl/binary.cpp:56]: op-plugin/op_plugin/OpApiInterface.h

Comment (hellozmz) [reply impl/ascend_npu/diopi_impl/binary.cpp:56]: div scalar 没有out版本

Comment (hellozmz) [reply impl/ascend_npu/diopi_impl/binary.cpp:56]: 使用 div_ 会导致修改了input

File: impl/ascend_npu/torch_npu/csrc/DIOPIAdapter.cpp [hunk 3]
```diff
@@ -1494,11 +1494,7 @@ at::Tensor OpPreparation::apply_tensor_with_sizes(c10::IntArrayRef sizes, const
 }
 
 at::Tensor OpPreparation::copy_scalar_to_device(const c10::Scalar& cpu_scalar, at::ScalarType scalar_data_type) {
-    at::Tensor cpu_tensor = scalar_to_tensor(cpu_scalar).to(scalar_data_type);
-    at::Tensor cpuPinMemTensor = cpu_tensor.pin_memory();
-    int deviceIndex = 0;
-    NPU_CHECK_ERROR(aclrtGetDevice(&deviceIndex));
-    return cpuPinMemTensor.to(c10::Device(c10::DeviceType::XLA, deviceIndex), cpuPinMemTensor.scalar_type(), true, true);
+    return CalcuOpUtil::CopyScalarToDevice(cpu_scalar, scalar_data_type);
```
Comment (POI-WX) [inline impl/ascend_npu/torch_npu/csrc/DIOPIAdapter.cpp:1497]: 这里应该不用再改动，直接调用会有未实现的接口。
image

**Review by yangbofun — APPROVED** [review]

**Pull request merged**